packages/amplify-codegen/tests/commands/models.test.js

```diff
@@ -0,0 +1,113 @@
+const generateModels = require('../../src/commands/models');
+const mock = require('mock-fs');
+const graphqlCodegen = require('@graphql-codegen/core');
+const fs = require('fs');
+const path = require('path');
+
+jest.mock('@graphql-codegen/core');
+const MOCK_CONTEXT = {
+  print: {
+    info: jest.fn(),
+  },
+  amplify: {
+    getProjectMeta: jest.fn(),
+    getEnvInfo: jest.fn(),
+    getResourceStatus: jest.fn(),
+    executeProviderUtils: jest.fn(),
+    pathManager: {
+        getBackendDirPath: jest.fn()
+    },
+    getProjectConfig: jest.fn()
+  },
+};
+const OUTPUT_PATHS = {
+    javascript: 'src',
+    android: 'app/src/main/java',
+    ios: 'amplify/generated/models',
+    flutter: 'lib/models'
+};
+const MOCK_PROJECT_ROOT = 'project';
+const MOCK_PROJECT_NAME = 'myapp';
+const MOCK_BACKEND_DIRECTORY = 'backend';
+const MOCK_GENERATED_CODE = 'This code is auto-generated!';
+
+// Mock the Feature flag to use migrated moldegen
+jest.mock('amplify-cli-core', (MOCK_PROJECT_ROOT) => {
+  return {
+    FeatureFlags: {
+      getBoolean: jest.fn().mockImplementation((name, defaultValue) => {
+        if (name === 'codegen.useappsyncmodelgenplugin') {
+          return true;
+        }
+      })
+    },
+    pathManager: {
+        findProjectRoot: jest.fn().mockReturnValue(MOCK_PROJECT_ROOT)
+    }
+  };
+});
+
+describe('command - models', () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+      MOCK_CONTEXT.amplify.getEnvInfo.mockReturnValue({projectPath: MOCK_PROJECT_ROOT});
+      MOCK_CONTEXT.amplify.getResourceStatus.mockReturnValue(
+           {
+              allResources: [
+                  {
+                      service: 'AppSync', 
+                      providerPlugin: 'awscloudformation',
+                      resourceName: MOCK_PROJECT_NAME
+                    }
+                ]
+            }
+        );
+      MOCK_CONTEXT.amplify.executeProviderUtils.mockReturnValue([]);
+      MOCK_CONTEXT.amplify.pathManager.getBackendDirPath.mockReturnValue(MOCK_BACKEND_DIRECTORY);
+      graphqlCodegen.codegen.mockReturnValue(MOCK_GENERATED_CODE);
+    });
+
+    for (const frontend in OUTPUT_PATHS) {
```

Comment:
Do we need the for loop for all frontend types? I think the tests can be simplified to one by mocking a frontend and its output dir and check if codegen function gets called.

Reply:
The function `getModelOutputPath` is not exported currently and is a private function, which prevents us from mocking it. The for loop was intended to avoid code duplication and test the model generation (mainly `loadSchema` and `getModelOutputPath`) for all front-ends.